okhttp-logging-interceptor/src/main/java/com/squareup/okhttp/logging/HttpLoggingInterceptor.java

```diff
@@ -179,7 +179,7 @@ public Level getLevel() {
       }
 
       String endMessage = "--> END " + request.method();
-      if (logBody && hasRequestBody) {
+      if (logBody && hasRequestBody && logBody(request.headers())) {
```

Comment:
I'm wondering if this should be wrapped around line 193 with an `else` to say "(gzip body omitted)" or something.